### ISSUE TITLE
Add *file* support

### ISSUE THREAD

**Issue by nate** [borkdude/babashka#179]
When writing scripts it's often convenient to colocate support files for that script. In my case I use a sync for my dotfiles and I'm using babashka to symlink those files into my home directory. Since the script and the dotfiles are colocated it's handy to be able to find the current source file's path. Since babashka is interpreted the source location of a file is a more reasonable thing to want.

I'd like to see something akin to joker's \*file\*, if possible. It doesn't resolve to anything when the context doesn't allow for it, like in a repl.

Thanks!

**Comment by borkdude** [comment]
How would you solve this problem if you were using normal JVM Clojure?

**Comment by nate** [comment]
I think I'd use `resource`? It's a scripting context so it feels like a little different problem space. Maybe we can adapt resource or something?

**Comment by borkdude** [comment]
@nate I mean, if you would have a `script.clj` that you would execute as `clojure script.clj`, it's a similar situation. What would you do there?

**Comment by borkdude** [comment]
A workaround hack for now:

Put this at the start of your script, e.g. `/tmp/script.clj`:
```
#!/bin/sh
#_(
   "exec" "bb" "$0" "$0" "hello" "$@"
   )
(prn *command-line-args*)
```

When calling `/tmp/script.clj`, it will print:

```
("/tmp/foo.clj" "hello")
```

Using normal `clojure` it seems you can get the current file's dir with:
```
(def x)
(require '[clojure.java.io :as io])
(def script-dir (.getParentFile (io/file (:file (meta #'x)))))
```
So it would maybe be good to implement that first.

**Comment by nate** [comment]
The workaround seems to work, thanks!

**Comment by j-cr** [comment]
Can't babashka just expose the :file from the parsed args as a var, e.g. `user/$0`?

**Comment by borkdude** [comment]
Snippet from Slack:

> Nate Sutton  @borkdude is that something I could work on, adding the :file? I might need a little direction to do it
and probably wouldn't be until tomorrow
borkdude  You could give it a try, but it requires some understanding of how files are processed either with passing a file argument, load-file and require. The way we handle classpath loading with a callback fn needs to change as well. I think it would be best if I gave it some more hammock time first
The logic is spread across babashka, the sci library and the edamame parser library
Each of these, or at least two, probably require changes.

**Comment by borkdude** [comment]
> Can't babashka just expose the :file from the parsed args as a var, e.g. user/$0?

Exposing it to the `user` namespace is probably not a good idea, since evaluating `*file*` in some other namespace will not work then. So it has to live in either `clojure.core` or in some other new global namespace like `babashka.core`.

**Comment by borkdude** [comment]
Fixed in 0.0.46:

https://github.com/borkdude/babashka/releases/tag/v0.0.46

**Comment by SeanSanker** [comment]
For anyone searching for the current docs on this feature they're located here: https://book.babashka.org/#_current_file_path
